common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -127,95 +127,96 @@ trait Logging {
-    if (log.isWarnEnabled) log.warn(msg)
-  }
-
-  protected def logWarning(entry: LogEntry): Unit = {
-    if (log.isWarnEnabled) {
-      withLogContext(entry.context) {
-        log.warn(entry.message)
-      }
-    }
-  }
+  protected def logDebug(msg: => String): Unit = if (log.isDebugEnabled) log.debug(msg)
+  protected def logError(msg: => String): Unit = if (log.isErrorEnabled) log.error(msg)
+  protected def logInfo(msg: => String): Unit = if (log.isInfoEnabled) log.info(msg)
+  protected def logTrace(msg: => String): Unit = if (log.isTraceEnabled) log.trace(msg)
+  protected def logWarning(msg: => String): Unit = if (log.isWarnEnabled) log.warn(msg)
+
+  // Log methods that take a string message and a Throwable (Exception/Error) too
+  protected def logDebug(msg: => String, throwable: Throwable): Unit =
+    if (log.isDebugEnabled) log.debug(msg, throwable)
+  protected def logError(msg: => String, throwable: Throwable): Unit =
+    if (log.isErrorEnabled) log.error(msg, throwable)
+  protected def logInfo(msg: => String, throwable: Throwable): Unit =
+    if (log.isInfoEnabled) log.info(msg, throwable)
+  protected def logTrace(msg: => String, throwable: Throwable): Unit =
+    if (log.isTraceEnabled) log.trace(msg, throwable)
+  protected def logWarning(msg: => String, throwable: Throwable): Unit =
+    if (log.isWarnEnabled) log.warn(msg, throwable)
 
-  protected def logWarning(entry: LogEntry, throwable: Throwable): Unit = {
-    if (log.isWarnEnabled) {
+  // Log methods that take only a LogEntry
+  protected def logDebug(entry: LogEntry): Unit = logDebug(entry, stripMargin = false)
+  protected def logError(entry: LogEntry): Unit = logError(entry, stripMargin = false)
+  protected def logInfo(entry: LogEntry): Unit = logInfo(entry, stripMargin = false)
+  protected def logTrace(entry: LogEntry): Unit = logTrace(entry, stripMargin = false)
+  protected def logWarning(entry: LogEntry): Unit = logWarning(entry, stripMargin = false)
+
+  // Log methods that take only a LogEntry and a Throwable (Exception/Error) too
+  protected def logDebug(entry: LogEntry, throwable: Throwable): Unit =
+    logDebug(entry, throwable, stripMargin = false)
+  protected def logError(entry: LogEntry, throwable: Throwable): Unit =
+    logError(entry, throwable, stripMargin = false)
+  protected def logInfo(entry: LogEntry, throwable: Throwable): Unit =
+    logInfo(entry, throwable, stripMargin = false)
+  protected def logTrace(entry: LogEntry, throwable: Throwable): Unit =
+    logTrace(entry, throwable, stripMargin = false)
+  protected def logWarning(entry: LogEntry, throwable: Throwable): Unit =
+    logWarning(entry, throwable, stripMargin = false)
+
+  // Log methods that take only a LogEntry with the option to strip margin from the string message
+  protected def logDebug(entry: LogEntry, stripMargin: Boolean): Unit =
+    logEntryHelper(entry, stripMargin, log.debug)
+  protected def logError(entry: LogEntry, stripMargin: Boolean): Unit =
```

Comment:
Adding a new parameter makes the framework more complicated. Shall we revert the API changes in this PR?

Reply:
@gengliangwang For parity with the previous logging, we need to call `stripMargin` on the resulting text log. For that we need the ability to tell the `logError` function to call `stripMargin` *after* we convert the `LogEntry` to a string. 

We can either (1) keep this API change, or (2) stop calling `stripMargin` on the logged strings. Which would you prefer? The use case comes from Catalyst rule logging here:

![image](https://github.com/apache/spark/assets/99207096/f9aa0719-5454-4b6e-8132-da62f40ec799)

Reply:
We talked offline and decided to add a `stripMargin` method to the `MessageWithContext` class for this purpose.